### ISSUE TITLE
Redundant else-clause -  but it's not

### ISSUE THREAD

**Issue by kroe761** [rubocop/rubocop#12993]
Rubocop is flagging an else statement where I am resetting a variable to nil as an redundant else statement
--------

## Expected behavior

I would not have expected to the `else` keyword to be flagged as a redundant  else.  That WOULD be true if I weren't trying to save the result of the conditional into a variable, but I am (hilariously, at the request of rubocop - it raised a Style/ConditionalAssignment warning). 

## Actual behavior

```
lib/my_code/reporting/reporting_formatter.rb:23:38: C: [Correctable] Style/EmptyElse: Redundant else-clause.
                                     else
                                     ^^^^
```

## Steps to reproduce the problem

```
@some_var = if array.size >= 2
              "Some value"
            else # rubocop:disable Style/EmptyElse
              nil
            end
```

## RuboCop version

```
1.62.1 (using Parser 3.3.0.5, rubocop-ast 1.31.2, running on ruby 3.1.2) [arm64-darwin23]
  - rubocop-performance 1.21.0

```

**Comment by Earlopain** [comment]
It is actually redundant:
```rb
a = 'a'
b = 'b'

a = if false
      'Some value'
    else
      nil
    end

b = if false
      'Some value'
    end

pp a
pp b
```

Both a and b are nil at the end. There is a config for this cop `EnforcedStyle: empty` which allows explicit nil but it is not enabled by default and only a stylistic impact

**Comment by kroe761** [comment]
Oh, I just assumed that if conditional wasn't executed the return variable would retain it's previous value.  TIL.  Thank you!
